tests/Aspire.Workload.Tests/WorkloadTestsBase.cs

```diff
@@ -0,0 +1,263 @@
+        }
+
+        return foundRows.ToArray();
+    }
+
+    // Don't fixup the prefix so it can have characters meant for testing, like spaces
+    public static string GetNewProjectId(string? prefix = null)
+        => (prefix is null ? "" : $"{prefix}_") + FixupSymbolName(Path.GetRandomFileName());
+
+    public static async Task<CommandResult?> AssertTestProjectRunAsync(string testProjectDirectory, string testType, ITestOutputHelper testOutput, string config = "Debug", int testRunTimeoutSecs = 3 * 60)
+    {
+        if (testType == "none")
+        {
+            Assert.False(Directory.Exists(testProjectDirectory), "Expected no tests project to be created");
+            return null;
+        }
+        else
+        {
+            Assert.True(Directory.Exists(testProjectDirectory), $"Expected tests project at {testProjectDirectory}");
+            using var cmd = new DotNetCommand(testOutput, label: $"test-{testType}")
+                                    .WithWorkingDirectory(testProjectDirectory)
+                                    .WithTimeout(TimeSpan.FromSeconds(testRunTimeoutSecs));
+
+            var res = (await cmd.ExecuteAsync($"test -c {config}"))
+                                .EnsureSuccessful();
+
+            Assert.Matches("Passed! * - Failed: *0, Passed: *1, Skipped: *0, Total: *1", res.Output);
+            return res;
+        }
+    }
+
+    internal static async Task AssertStarterTemplateRunAsync(IBrowserContext context, AspireProject project, string config, ITestOutputHelper _testOutput)
+    {
+        await project.StartAppHostAsync(extraArgs: [$"-c {config}"], noBuild: false);
+
+        var page = await project.OpenDashboardPageAsync(context);
+        ResourceRow[] resourceRows;
+        try
+        {
+            resourceRows = await CheckDashboardHasResourcesAsync(
+                                    page,
+                                    StarterTemplateRunTestsBase<StarterTemplateFixture>.GetExpectedResources(project, hasRedisCache: false),
+                                    _testOutput).ConfigureAwait(false);
+        }
+        catch
+        {
+            string screenshotPath = Path.Combine(project.LogPath, "dashboard-fail.png");
+            await page.ScreenshotAsync(new PageScreenshotOptions { Path = screenshotPath });
+            _testOutput.WriteLine($"Dashboard screenshot saved to {screenshotPath}");
+            throw;
+        }
+
+        string url = resourceRows.First(r => r.Name == "webfrontend").Endpoints[0];
+        await StarterTemplateRunTestsBase<StarterTemplateFixture>.CheckWebFrontendWorksAsync(context, url, _testOutput, project.LogPath);
+        await project.StopAppHostAsync();
+    }
+
+    private static readonly char[] s_charsToReplace = ['.', '-', '+'];
+
+    public static string FixupSymbolName(string name)
+    {
+        UTF8Encoding utf8 = new();
```

Comment:
We shouldn't need to encode the name to UTF8 just to do this fix up / normalizing.